File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        String vowels = "aeiouAEIOU"; 
        char[] arr = s.toCharArray();
        int l = 0, r = arr.length - 1;

        while (l < r) {
            while (l < r && vowels.indexOf(arr[l]) == -1) l++; // Move left pointer
            while (l < r && vowels.indexOf(arr[r]) == -1) r--; // Move right pointer

            char temp = arr[l];
            arr[l] = arr[r];
            arr[r] = temp;

            l++;
            r--;
        }

        return new String(arr);
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, sentence: str) -> str:
        s =  list(sentence)
        vow = "aeiouAEIOU"
        l = 0
        r = len(s) -1
        while l < r:
            if s[l] in vow and s[r] in vow:
                s[l],s[r] = s[r],s[l]
                l+=1
                r-=1
            elif s[l] not in vow:
                l+=1
            else:
                r-=1
        return "".join(s)